test/document_stores/test_pinecone.py

```diff
@@ -471,6 +471,36 @@ def test_get_embedding_count(self, doc_store_with_docs: PineconeDocumentStore):
         doc_store_with_docs.write_documents([doc])
         assert doc_store_with_docs.get_embedding_count() == 1
 
+    @pytest.mark.integration
+    def test_get_document_count(self, doc_store_with_docs: PineconeDocumentStore):
+        """
+        Tests that get_document_count() returns the correct number of documents in the document store.
+        We use get_document_count(only_documents_without_embedding=True) to check that the number of documents without
+        embeddings is correct and get_document_count() to check that the total number of documents is correct.
+        """
+        # 9 documents without embeddings are already in doc_store_with_docs, so we expect 9 documents without embeddings
+        assert doc_store_with_docs.get_document_count(only_documents_without_embedding=True) == 9
+
+        # document with embedding is written to doc_store_with_docs
+        doc = Document(content=f"Doc with embedding", embedding=np.random.rand(768).astype(np.float32))
+        doc_store_with_docs.write_documents([doc])
+
+        # 9 documents without embeddings are already in doc_store_with_docs, so we expect 10 documents in total
+        assert doc_store_with_docs.get_document_count() == 10
+
+        # but we expect 9 documents without embeddings to be unchanged
+        assert doc_store_with_docs.get_document_count(only_documents_without_embedding=True) == 9
+
+        # another document with embedding is written to doc_store_with_docs
+        doc_2 = Document(content=f"Doc_2 with embedding", embedding=np.random.rand(768).astype(np.float32))
+        doc_store_with_docs.write_documents([doc_2])
+
+        # we now expect 11 documents in total
+        assert doc_store_with_docs.get_document_count() == 11
+
+        # but we expect documents without embeddings to remain at 9
+        assert doc_store_with_docs.get_document_count(only_documents_without_embedding=True) == 9
+
```

Comment:
I think this is a bit overly complex. Can we make a unit test for this change, mocking Pinecone?

Reply:
I've broken down unit tests on this one, can please you have a quick look again @ZanSara?